pyg_lib/csrc/segment/cuda/matmul_kernel.cu

```diff
@@ -0,0 +1,157 @@
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <torch/library.h>
+
+#include <cutlass/gemm/device/gemm_grouped.h>
+#include <cutlass/gemm/kernel/default_gemm_grouped.h>
+#include <cutlass/util/host_tensor.h>
+
+namespace pyg {
+namespace segment {
```

Comment:
My feeling is that `pyg::segment` might better be something like `pyg::ops::segment`. Want to hear about how you would organize future kernels